#! /bin/sh /usr/share/dpatch/dpatch-run
## 18-fix-drawichar.dpatch by Michael Karcher
## <debian@mkarcher.dialup.fu-berlin.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix cast for indexing an array with a character.

@DPATCH@
diff -urNad rott-1.0+dfsg~/rt_str.c rott-1.0+dfsg/rt_str.c
--- rott-1.0+dfsg~/rt_str.c	2008-01-09 13:52:29.000000000 +0100
+++ rott-1.0+dfsg/rt_str.c	2008-01-10 09:19:19.000000000 +0100
@@ -1517,8 +1517,8 @@
    dest = origdest;
 
    ch -= 31;
-   width = IFont->width[ (unsigned int)ch ];
-   source = ( ( byte * )IFont ) + IFont->charofs[ (unsigned int)ch ];
+   width = IFont->width[ (unsigned char)ch ];
+   source = ( ( byte * )IFont ) + IFont->charofs[ (unsigned char)ch ];
 
    mask = 1 << ( px & 3 );
 
